torchgeo/datasets/utils.py

```diff
@@ -598,6 +600,25 @@ def unbind_samples(sample: Dict[Any, Sequence[Any]]) -> List[Dict[Any, Any]]:
     return _dict_list_to_list_dict(sample)
 
 
+def rearrange_patches_to_sample(batch: List[Dict[str, Any]]) -> Dict[str, Any]:
+    """Define collate function to combine patches per tile and batch size.
+
+    Args:
+        batch: sample batch from dataloader containing image and mask
+
+    Returns:
+        sample batch where the batch dimension is
+        'batch_size' * 'num_patches_per_tile'
+
+    .. versionadded: 0.4
+    """
+    r_batch: Dict[str, Any] = default_collate(batch)  # type: ignore[no-untyped-call]
+    if "mask" in r_batch:
+        r_batch["mask"] = rearrange(r_batch["mask"], "b t h w -> (b t) h w")
+    r_batch["image"] = rearrange(r_batch["image"], "b t c h w -> (b t) c h w")
```

Comment:
Should also check if "image" is in dict. Actually, could we just loop over all tensors in the dict and merge the first two dims? Then we don't have to duplicate code or worry about names.

Reply:
Another option (in terms of both code and function name) would be `torch.flatten(value, end_dim=1)`

Reply:
> Should also check if "image" is in dict. Actually, could we just loop over all tensors in the dict and merge the first two dims? Then we don't have to duplicate code or worry about names.

Is there a scenario where we use this particular function and "image" is not present in the dict? 


Regarding the code. I like the current einops implementation as it's more clear to the user the what the operation does to the image (5D -> 4D) and mask tensors (4D -> 3D). Even if they're undergoing the same op i.e. merging first two dims.

Reply:
I think I also find the more explicit einops operation better because it would be more clearer if someone looks at what `rearrange_samples` does in the source code.

Reply:
> Is there a scenario where we use this particular function and "image" is not present in the dict?

Most GeoDatasets contain either an image or a mask but not both. So any time someone used this collation function for a mask-only dataset this would break.

Reply:
The problem with this is that it makes assumptions about the keys of the dict. None of our other collation functions do this, and I don't think it's necessary or wise to do this. For example, what if someone wants to use this with 6D images or 5D masks?

Reply:
For example, additional axis could be time for time-series data, or z for x,y,z climate/weather data.